### PR TITLE
Deploy on push/k8s/test purging cidr

### PR DESCRIPTION
**Pull request by andreic92**
~~The changes are required in order to be able to purge (or banning) resources right after these are stored to UPP.
The guidelines for the changes are described [here](https://docs.google.com/a/ft.com/document/d/1OlLo4PLJYwiku9RpQT-r-J0fMKsyOJlzAZmAonJWXmQ/edit?disco=AAAAB2hABNw)
The Jira issue describing the changes can be found [here](https://jira.ft.com/browse/FDC-91).
The changes are inspired by [this](https://varnish-cache.org/docs/trunk/users-guide/purging.html#bans) documentation.~~

EDIT:
The changes are required in order to be able to purge (or banning) resources right after these are stored to UPP.
The guidelines for the changes are described in [here](https://docs.google.com/a/ft.com/document/d/1OlLo4PLJYwiku9RpQT-r-J0fMKsyOJlzAZmAonJWXmQ/edit?disco=AAAAB2hABNw)
The Jira issue describing the changes can be found in [here](https://jira.ft.com/browse/FDC-91).
The changes are inspired by [this](https://varnish-cache.org/docs/trunk/users-guide/vcl-hashing.html) documentation.

The concern in here is that the changes could have a negative impact on the current state of the UPP. Please take a look and if you can spot something that could be harmful, please let me know. 

Initially (for this PR) was about banning mechanism integration. Meanwhile we've found out why purging hasn't worked as expected therefore the current PR contains changes on how the caching is performed in order to tune it for better performance and easier purging requests.

Before this PR, the caching was performed using the default hashing strategy
```
sub vcl_hash {
    hash_data(req.url);
    if (req.http.host) {
        hash_data(req.http.host);
    } else {
        hash_data(server.ip);
    }
    return (lookup);
}
```
and after this PR the hashing will be changed to
```
sub vcl_hash {
    hash_data(std.tolower(req.url));
    return (lookup);
}
```
In simple words, the cache key of the entry will be a hash created only with the url instead being created with the url + (hostname/ip)

~~The changes are inspired by [this](https://varnish-cache.org/docs/trunk/users-guide/purging.html#bans) documentation.~~

### CODE REVIEW

File: default.vcl [hunk 1]
```diff
@@ -68,14 +68,23 @@ sub vcl_recv {
     # Remove all cookies; we don't need them, and setting cookies bypasses varnish caching.
     unset req.http.Cookie;
 
-    # allow PURGE from localhost
+    # allow PURGE from localhost and 10.2...
     if (req.method == "PURGE") {
```
Comment (tamas-molnar) [inline default.vcl]: Do we still need purge if we have ban?

Comment (andreic92) [reply default.vcl]: The purge block was in there before my changes. The only change I've made regarding the purging code was to enhance the `acl purge` in order to allow access from the pods inside the cluster. But this `acl purge` is also used for the banning block.

Comment (tamas-molnar) [reply default.vcl]: :+1:

**Comment by peterschubert** [comment]
If we exclude the host from the hash, we might run the risk of undesired hash collisions between homonymous endpoints of different services. I don't really know if that is the case now or a risk for the future.

**Comment by yolandeleungFT** [comment]
I cannot think of any requests we currently have where we would vary the host and have the same path in the same varnish... Or envisage that we will have soon.

However, I wonder why we have the host here as part of the cache key? Is the Host header and therefore the client host?

**Pull request merged**